lms/djangoapps/course_structure_api/v0/tests.py

```diff
@@ -213,21 +216,20 @@ def test_not_authorized(self):
 class CourseListTests(CourseViewTestsMixin, SharedModuleStoreTestCase):
     view = 'course_structure_api:v0:list'
 
-    @classmethod
-    def setUpClass(cls):
-        super(CourseListTests, cls).setUpClass()
-        cls.create_course_data()
-
     def test_get(self):
         """
         The view should return a list of all courses.
         """
-        response = self.http_get(reverse(self.view))
+        with check_mongo_calls(0):
+            response = self.http_get(reverse(self.view))
         self.assertEqual(response.status_code, 200)
         data = response.data
         courses = data['results']
-        self.assertEqual(len(courses), 2)
-        self.assertEqual(data['count'], 2)
+
+        # an extra course (course-v1:org.0+course_0+Run_0) appears on the testing environment
```

Comment:
BTW @dsjen I am able to reproduce locally if I run the full shard-2 for lms. It fails somewhat early on... but the command I use is `paver test_system -s lms --extra_args="--attr='shard_1=False' --with-flaky" --cov_args="-p"` which is what Jenkins is using on that shard (it's in the generic tests shell script under the scripts directory).